terragrunt/aws/alert_compromise/functions/broadcast_alert.py

```diff
@@ -0,0 +1,38 @@
+import json
+import boto3
+import gzip
+import base64
+import os
+import re
+
+#===============================================================================
+
+def lambda_handler(event, context):
+    """
+    This function is triggered by a CloudWatch Logs subscription filter and sends an alert to an SNS topic if a secret is detected in the logs.
+
+    Args:
+        event (dict): The event data passed by AWS Lambda.
+        context (object): The context object passed by AWS Lambda.
+
+    Returns:
+        None
+    """
+    decoded_event = json.loads(gzip.decompress(base64.b64decode(event['awslogs']['data'])))
+    message = decoded_event['logEvents'][0]['message']
+    # Double check that the message received contains a secret
+    if ("Secret detected:" in message ):
```

Comment:
Just a minor comment that because the CloudWatch filter event is only invoking the function on this `Secret detected` pattern, you could get rid of this `if`, but it doesn't hurt to have it.